### PR TITLE
Update Coverage.py 

### LINKED ISSUE #1898: Cython.Coverage: error with pyx-file coverage with lower-case filenames and wrong line numbers in .coverage
On windows, 
with cython0.27
and 
Coverage.py, version 4.3.4 with C extension

I try to report the coverage of a pyx-file with the Cython.Coverage plugin for coverage.py
`coverage run --source=src -m py.test --pyargs cythonarrays` the .coverage-file including line-numbers that were hit for the .pyx files.
However, `coverage report` issues 

> src\cythonarrays\array_shapes.pyx   TypeError: 'NoneType' object is not iterable
> src\cythonarrays\tests\example_cython.pyx   TypeError: 'NoneType' object is not iterable

Unfortunately, the coverage module catches the exception and did not show where the error occured.
So i had to patch coverage.summary.py to get the tracebak, which showed me, that set(self._code), becuase self._code was None in line 286 in 

```
class CythonModuleReporter(FileReporter):
    def lines(self):
        return set(self._code)
```

Further debugging led me to the point, that the "None" of self._code came from

```
class Plugin(CoveragePlugin):
    def _parse_lines(self, c_file, sourcefile):
        if sourcefile not in self._c_files_map:
            return (None,) * 2  # e.g. shared library file
```

and sourcefile was not in self._c_files_map, because sourcefile started with C:\Dev\cythonarrays\..., while the key in the self._c_files_map-dictionary was written with lowercase c:\dev\cythonarrays...
So abspath returns the path with lowercase because it finds the files in PATH. I installed the package ediable with pip install -e cythonarrays and this lead to entries in easyinstall.pth with lowercase path.

To patch the problem, i converted all filepaths to lowercase

```
class Plugin(CoveragePlugin):

    def file_reporter(self, filename)
        filename = os.path.abspath(filename).lower()

    def _parse_lines(self, c_file, sourcefile):
        (...)
        for filename, code in code_lines.items():
            abs_path = _find_dep_file_path(c_file, filename).lower()
```
This adds the *.pyx-files to the report.

However, this might be not the best solution, which works always on windows and linux...
Maybe you have another idea how to fix this.

The other problem are the line numbers reported by `coverage run` and stored in the .coverage result file, for the .pyx-file, which do not match the actual line numbers in the pyx-file.
In the .coverage -file, for my example_cython.pyx-file there are 26 line numbers between 1 and 983 collected.
`!coverage.py: This is a private format, don't read it directly!{"lines":{"C:\\Dev\\cythonarrays\\code\\cythonarrays\\src\\cythonarrays\\tests\\example_cython.pxd":[],"C:\\Dev\\cythonarrays\\code\\cythonarrays\\src\\cythonarrays\\tests\\example_cython.pyx":[1,246,10,288,207,983,537,155,284,285,286,31,32,289,290,291,36,37,287,243,244,245,54,55,314,315]},"file_tracers":{"C:\\Dev\\cythonarrays\\code\\cythonarrays\\src\\cythonarrays\\tests\\example_cython.pyx":"Cython.Coverage.Plugin","C:\\Dev\\cythonarrays\\code\\cythonarrays\\src\\cythonarrays\\tests\\example_cython.pxd":"Cython.Coverage.Plugin","C:\\Dev\\cythonarrays\\code\\cythonarrays\\src\\cythonarrays\\array_shapes.pyx":"Cython.Coverage.Plugin"}}`

 The  example_cython.pyx file has only 86 lines. 

The C-File i generated contains comments for the traced lines in the actual .pyx-file like:

>  /* "cythonarrays/tests/example_cython.pyx":31
 
to C-imported files like:
 

> /* "../../../../Anaconda/envs/cythonarrays/lib/site-packages/Cython/Includes/numpy/__init__.pxd":220

and to View.MemoryView and to a "Tree fragment"

> /* "View.MemoryView":120
>  /* "(tree fragment)":1

So the strange numbers the Cython.Coverage plugin collected in the c-file are mostly line numbers refering to the "View.MemoryView" 

```
  /* "View.MemoryView":983
 *         return self.from_object
 * 
 *     __pyx_getbuffer = capsule(<void *> &__pyx_memoryview_getbuffer, "getbuffer(obj, view, flags)")             # <<<<<<<<<<<<<<
 * 
 * 
 */
```

or to the "tree fragment"

 ```
 /* "(tree fragment)":1
 * def __reduce_cython__(self):             # <<<<<<<<<<<<<<
 *     cdef bint use_setstate
 *     state = (self.INF_d, self.INF_f, self.NAN_d, self.NAN_f, self.NINF_d, self.NINF_f, self._invalid_g, self._jobs_j, self._km_ij, self._not_initialized_ij, self._param_g, self._persons_gi, self._trips_ij, self._valid_g, self._zonenumbers_i, self.destinations, self.groups, self.n_threads, self.origins)
 */
```
What could be the reason that these line numbers are collected instead of the numbers of the lines in the actual .pyx-file, that were hit?

### PR DESCRIPTION
**Pull request by MaxBo**
this commit fixes #1898
use filepath with the correct upper and lowercases

### CODE REVIEW

File: Cython/Coverage.py [hunk 1]
```diff
@@ -45,6 +46,35 @@ def _find_dep_file_path(main_file, file_path):
...
+    -------
+    filepath : str
+    """
+    if not sys.platform.startswith('win'):
+        return filepath
+    dirs = filepath.split('\\')
+    # disk letter
+    test_name = [dirs[0].upper()]
+    for d in dirs[1:]:
+        test_name += ["%s[%s]" % (d[:-1], d[-1])]
```
Comment (scoder) [inline Cython/Coverage.py]: How to append to a list? Easy! Create a new one and concatenate it to the existing one! :D
(I know, it's now your code...)

File: Cython/Coverage.py [hunk 2]
```diff
@@ -45,6 +46,35 @@ def _find_dep_file_path(main_file, file_path):
     return abs_path
 
 
+def _get_actual_filename(filepath):
```
Comment (scoder) [inline Cython/Coverage.py]: I think this should just use `os.path.normcase()`.

File: Cython/Coverage.py (hunk 1 above)
Comment (MaxBo) [reply Cython/Coverage.py]: Thats whats happes if you copy&paste random code ;-)
I fixed this

**Comment by MaxBo** [comment]
Fixed coverage tests in windows by renaming the generated pyd-files from `pkg\\coverage_test_py.cp36-win_amd64.pyd` to `pkg\\coverage_test_py.pyd`

Added MixedCase file- and packagenames to the coverage_api tests

File: Cython/Coverage.py [hunk 3]
```diff
@@ -9,9 +9,11 @@
 import re
 import os.path
 import sys
+import glob
```
Comment (scoder) [inline Cython/Coverage.py]: Left-over import.

Comment (MaxBo) [reply Cython/Coverage.py]: deleted

**Pull request merged**